### DISCUSSION TITLE
Terminal Support

### DISCUSSION THREAD

**Discussion by AnonymouX47** [AnonymouX47/term-image#4] [General]
Report names (and versions, if applicable) of terminal emulators yet to be listed [here]() that have been tested to meet all requirements and work properly with ``term-image``.

Also, report terminal-specific issues.

Thanks!

**Comment by AbeEstrada** [comment]
I'm running the test on `('ghostty', '0.1.0-main+955246d6')`, and `term-image` passes all the tests except for `test-renderable-renderable`. It hangs at `tests/renderable/test_renderable.py::TestDraw::TestNonAnimation::TestEchoInput::test_default`. However, I also ran the same test using `('wezterm', '20240203-110809-5046fc22')`, and it hangs too, so I think it is safe to list Ghostty as a compatible terminal, but I understand if you want to wait until you have access to it and can run the tests.

I know it is in 'invite-only' beta for now. There will eventually be an open beta period, but if you want to get an invite, please go to the [Ghostty](https://github.com/ghostty-org) Discord channel. There are waves of invites from time to time.

**Reply by AnonymouX47** [reply]
That's quite an interesting issue... I suspect it has something to do with your platform or environment.

If you don't mind:

- What's your OS (and distro if on GNU/Linux) and version?
- Is the OS running on bare metal or in a VM?
- What Python version are you running the tests with?
- How well do other `termios`-based software behave within the same environment on the same machine?

Also, can you please run:

```py
pytest -vv -r a --full-trace tests/renderable/test_renderable.py
```
Then hit `Ctrl-C` while it's hung and share the traceback.

Thank you :smiley:.

EDIT: As for ghostty, I'll see what I can do. Thanks.

**Reply by AbeEstrada** [reply]
I submitted an issue https://github.com/AnonymouX47/term-image/issues/110
